src/ImmersedBoundaries/immersed_grid_metrics.jl

```diff
@@ -34,6 +36,11 @@ end
 @inline Δzᵃᵃᶠ(i, j, k, ibg::IBG) = Δzᵃᵃᶠ(i, j, k, ibg.underlying_grid)
 
 return_metrics(grid::IBG) = return_metrics(grid.underlying_grid)
-xspacings(X, grid::IBG) = xspacings(X, grid.underlying_grid)
-yspacings(Y, grid::IBG) = yspacings(Y, grid.underlying_grid)
-zspacings(Z, grid::IBG) = zspacings(Z, grid.underlying_grid)
+
+xspacing(grid::IBG, args...) = xspacing(grid.underlying_grid, args...)
+yspacing(grid::IBG, args...) = yspacing(grid.underlying_grid, args...)
+zspacing(grid::IBG, args...) = zspacing(grid.underlying_grid, args...)
+
+xspacings(grid::IBG, args...) = xspacings(grid.underlying_grid, args...)
+yspacings(grid::IBG, args...) = yspacings(grid.underlying_grid, args...)
+zspacings(grid::IBG, args...) = zspacings(grid.underlying_grid, args...)
```

Comment:
this does not pass kwargs, e.g.,

```Julia
julia> xspacings(ibg, Center(); with_halos=false)
ERROR: MethodError: no method matching xspacings(::ImmersedBoundaryGrid{Float64, Periodic, Periodic, Bounded, RectilinearGrid{Float64, Periodic, Periodic, Bounded, Float64, Float64, Float64, OffsetArrays.OffsetVector{Float64, StepRangeLen{Float64, Base.TwicePrecision{Float64}, Base.TwicePrecision{Float64}, Int64}}, OffsetArrays.OffsetVector{Float64, StepRangeLen{Float64, Base.TwicePrecision{Float64}, Base.TwicePrecision{Float64}, Int64}}, OffsetArrays.OffsetVector{Float64, StepRangeLen{Float64, Base.TwicePrecision{Float64}, Base.TwicePrecision{Float64}, Int64}}, CPU}, GridFittedBottom{OffsetArrays.OffsetMatrix{Float64, Matrix{Float64}}, Oceananigans.ImmersedBoundaries.CenterImmersedCondition}, Nothing, CPU}, ::Center; with_halos::Bool)

Closest candidates are:
  xspacings(::ImmersedBoundaryGrid, ::Any...) got unsupported keyword argument "with_halos"
   @ Oceananigans ~/Research/OC5.jl/src/ImmersedBoundaries/immersed_grid_metrics.jl:44
  xspacings(::Any, ::Any, ::Any, ::Any; with_halos)
   @ Oceananigans ~/Research/OC5.jl/src/Grids/grid_utils.jl:363
  xspacings(::OrthogonalSphericalShellGrid, ::Center, ::Center; with_halos)
   @ Oceananigans ~/Research/OC5.jl/src/Grids/orthogonal_spherical_shell_grid.jl:955
  ...

Stacktrace:
 [1] top-level scope
   @ REPL[17]:1
```

I think this will do the job 

```suggestion
xspacings(grid::IBG, args...; kwargs...) = xspacings(grid.underlying_grid, args...)
yspacings(grid::IBG, args...; kwargs...) = yspacings(grid.underlying_grid, args...)
zspacings(grid::IBG, args...; kwargs...) = zspacings(grid.underlying_grid, args...)
```

Reply:
Thanks!